### ISSUE TITLE
xDrip Updater not detecting new releases

### ISSUE THREAD

**Issue by Lupum1001** [NightscoutFoundation/xDrip#2146]
I'm currently running the build from **Oct 4, 2021** and I can't get xDrip to detect any new releases from the Stable/Beta/Alpha channels.
In theory, it _should_ detect the build from Jan 31st, but it currently only can detect builds from the nightly channel.

Link to discussion thread: https://github.com/NightscoutFoundation/xDrip/discussions/2145

---

### My Environment:
Samsung Galaxy S21 phone running xDrip version:
**49817a5-2021.10.04**
Code: 2110041402

### Expected behavior
When I check for an update, either by switching channels (which triggers a check), or pressing the explicit "check for updated version" menu item, I'd expect to see the usual pop-up offering an update to the Jan 31st build.

### Actual behavior
When I've selected the Stable, Beta, or Alpha channels, I don't get a pop-up. This is also true if I explicitly press the "check for updated version" button in the Home screen dropdown menu.
It _will_ detect the nightly builds, if I select that channel, but that's it.

**Comment by Navid200** [comment]
I would change the title to "xDrip updater only works for the Nightly channel".
But, that's just a suggestion.  Keep it if you like.

**Comment by Navid200** [comment]
@jamorham 
1- I remember people getting updates from the beta channel from posts in facebook in 2021.
2- I installed versions all the way back to September 2020 and they all have this problem.

Putting 1 and 2 together, the problem cannot be in the code.  It must be with the links.  But, I don't understand what.  I'm not sure what to look for.

**Comment by Lupum1001** [comment]
Looking at the code, My first instinct is an integer overflow.

In ascending order:
2110041402 - My installed version
2122051908 - latest from nightly channel (not sure why this doesn't start with a 22, tbh)
2147483647 - **maximum value of signed 32-bit integer**
2201310800 - latest from Stable/Beta/Alpha channels

![image](https://user-images.githubusercontent.com/52087143/170394595-ca2b8873-d854-4850-9d82-d1a12e7d1137.png)

**Comment by Lupum1001** [comment]
Looks like @jamorham patched the version numbers on January 3rd: https://github.com/NightscoutFoundation/xDrip/commit/666f5788d96997d0b30bf1a960a4a9f049066c2c

It seems to be working for the latest nightly builds, and the January release itself is actually fine (if you install the apk, it correctly self-identifies as 2122013118).
The January 31st build on the website is just labeled as 2201310800 and needs to be corrected this once.

**Comment by Navid200** [comment]
@Lupum1001 Thanks for all your work.
He is sleep now.  I'm sure he will sort it out.

**Comment by jamorham** [comment]
Thanks for noticing this. Please can anyone who is interested check if the alpha channel is now working correctly? Thanks.

**Comment by Lupum1001** [comment]
👍 Yes, Alpha is now working for me. It shows version 2122051908 (May 19th build)

**Comment by Navid200** [comment]
Yes, alpha is working pointing to a very very recent Nightly, the current one.

**Comment by jamorham** [comment]
I've promoted the currently nightly to alpha which is why it shows as the May 19th one.

**Comment by Lupum1001** [comment]
> https://xdrip-plus-updates.appspot.com/update-check/stable
> 
> ```
> 2201310800
> https://github.com/NightscoutFoundation/xDrip/releases/download/2022.01.31/xDrip-plus-20220131-33be05a.apk?
> 12779931
> 
> c28cc7f8786c7147876caf0cd869af33e0a8e09a4f4bfc8ffe7e2c1456bd36db
> ```
> 
> https://xdrip-plus-updates.appspot.com/update-check/nightly
> 
> ```
> 2122051908
> https://github.com/NightscoutFoundation/xDrip/releases/download/2022.05.19/xDrip-plus-20220519-e557cba.apk?
> 13050888
> 
> 44b24ed125a1425b9497af9136e31355e1ec9ec8dc477110ec8c2572685bce19
> ```

Is it possible to update the version # of the January release on appspot?
As I mentioned earlier, the _file_ already thinks it's 2122013118, so it's just appspot's metadata for that file causing issues.

**Comment by Navid200** [comment]
On Monday, June 13 2022, the Beta and Stable channels will change from January 31, 2022, to May 19, 2022.  

  
If you are on the Beta or Stable channels, you may get a notification that a new update is available for you.
You can go ahead and download and update.  
  
Do you have to update?
You never have to update. But, you can.
  

You can read about how update channels work here: [https://navid200.github.io/xDrip/docs/Updates.html](https://navid200.github.io/xDrip/docs/Updates.html)

**Comment by Navid200** [comment]
All update channels are working fine now.

We can reopen if you are still experiencing the problem.
